sys/net/application_layer/dhcpv6/client.c

```diff
@@ -931,4 +935,34 @@ static void _rebind(event_t *event)
     _request_renew_rebind(DHCPV6_REBIND);
 }
 
+static void _set_event_timeout_ms(event_timeout_t *timeout, event_t *event,
+                                  uint32_t delay_ms)
+{
+#if IS_USED(MODULE_EVENT_TIMEOUT_ZTIMER)
+    event_timeout_ztimer_init(timeout, ZTIMER_MSEC, event_queue, event);
+    event_timeout_set(timeout, delay_ms);
+#else
+    event_timeout_init(timeout, event_queue, event);
+    event_timeout_set(timeout, delay_ms * US_PER_MS);
+#endif
+}
+
+static void _set_event_timeout_sec(event_timeout_t *timeout, event_t *event,
+                                   uint32_t delay_sec)
+{
+#if IS_USED(MODULE_EVENT_TIMEOUT_ZTIMER)
+    event_timeout_ztimer_init(timeout, ZTIMER_SEC, event_queue, event);
+    event_timeout_set(timeout, delay_sec);
+#else
+    event_timeout_init(timeout, event_queue, event);
+    /* use xtimer_set64 instead of event_timeout_set to prevent overflows */
+    xtimer_set64(&timeout->timer, ((uint64_t)delay_sec) * US_PER_SEC);
+#endif
```

Comment:
> Maybe i would like the use of just msec (one ztimer to rule them all, this could also avoid some jitter but dhcpv6 seems to be tolerant to that) question

Main reason for also using `ZTIMER_SEC` is this line. I really do not want to cap of a potential overflow here. Since this function is used to set the T1 and T2 timeouts (both come over the wire) it is not guaranteed that they will be e.g. `UINT32_MAX - 1`.

Reply:
This may still be problematic since gnrc_dhcp `dhcpv6_client_conf_prefix` and nib `_nib_pl_add` have a max addr lifetime of uint32_max-1 msecs or infite
which will make the addr last shorter that its T1 and T2 timeout

Reply:
Sure, but nothing prevents us to port that to `ZTIMER_SEC` in the future as well ;-). Currently the same problem exists in master, since the `xtimer` of DHCPv6 uses 64-bit scale.